q2templates/templates/js-error-handler.html

```diff
@@ -0,0 +1,24 @@
+{# This block is part of it's own script tag, that way it is parsed and #}
+{# executed before any potentially buggy code. When using this, order #}
+{# matters - try to include this as high up as possible in the calling #}
+{# template, that way this handler can catch as much as possible. #}
+<script type="text/javascript">
+  function handleErrors(errors, element, msg) {
+    console.error(errors);
+    msg = msg === undefined ? '' : msg + '<br><br>';
+    var helpMsg = '<p class="alert alert-danger"> \
+                   There was an error loading and rendering the \
+                   visualization.<br><br>' + msg + 'Please \
+                   copy-and-paste the following error messages when \
+                   reporting this error (please be aware that the content \
+                   of your analysis data and/or metadata is included in \
+                   these messages):</p>';
+    var helpMsgDiv = $('<div></div>');
+    helpMsgDiv.append(helpMsg);
+    helpMsgDiv.append('<pre><code>' + errors.join('\n\n//----\n\n') + '</code></pre>');
```

Comment:
This looks a little weird:
```
//----
```
What's the goal of it?

Reply:
Oh I get it now. Would a double newline suffice here?

Reply:
How about this:

![screen shot 2018-07-19 at 4 23 25 pm](https://user-images.githubusercontent.com/274668/42975265-1ab3ec82-8b70-11e8-9c89-50f93d5bddfe.png)

Reply:
lgtm!